File: app/src/main/java/com/example/khughes/machewidget/CarStatus/WindowPosition.java
package com.example.khughes.machewidget.CarStatus;

import androidx.room.Embedded;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import javax.annotation.Generated;

@Generated("jsonschema2pojo")
public class WindowPosition {

    @SerializedName("driverWindowPosition")
    @Expose
    @Embedded
    private DriverWindowPosition driverWindowPosition;
    @SerializedName("passWindowPosition")
    @Expose
    @Embedded
    private PassWindowPosition passWindowPosition;
    @SerializedName("rearDriverWindowPos")
    @Expose
    @Embedded
    private RearDriverWindowPos rearDriverWindowPos;
    @SerializedName("rearPassWindowPos")
    @Expose
    @Embedded
    private RearPassWindowPos rearPassWindowPos;

    public DriverWindowPosition getDriverWindowPosition() {
        return driverWindowPosition;
    }

    public void setDriverWindowPosition(DriverWindowPosition driverWindowPosition) {
        this.driverWindowPosition = driverWindowPosition;
    }

    public PassWindowPosition getPassWindowPosition() {
        return passWindowPosition;
    }

    public void setPassWindowPosition(PassWindowPosition passWindowPosition) {
        this.passWindowPosition = passWindowPosition;
    }

    public RearDriverWindowPos getRearDriverWindowPos() {
        return rearDriverWindowPos;
    }

    public void setRearDriverWindowPos(RearDriverWindowPos rearDriverWindowPos) {
        this.rearDriverWindowPos = rearDriverWindowPos;
    }

    public RearPassWindowPos getRearPassWindowPos() {
        return rearPassWindowPos;
    }

    public void setRearPassWindowPos(RearPassWindowPos rearPassWindowPos) {
        this.rearPassWindowPos = rearPassWindowPos;
    }

}

File: app/src/main/java/com/example/khughes/machewidget/CarStatus/WindowPosition.kt
package com.example.khughes.machewidget.CarStatus

import androidx.room.ColumnInfo
import androidx.room.Embedded
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import javax.annotation.Generated

@Generated("jsonschema2pojo")
class WindowPosition {

    @Generated("jsonschema2pojo")
    class DriverWindowPosition {
        @SerializedName("value")
        @Expose
        @ColumnInfo(name = "driverwindow_value")
        var value: String? = null
    }

    @Generated("jsonschema2pojo")
    class PassWindowPosition {
        @SerializedName("value")
        @Expose
        @ColumnInfo(name = "passwindow_value")
        var value: String? = null
    }

    @Generated("jsonschema2pojo")
    class RearDriverWindowPos {
        @SerializedName("value")
        @Expose
        @ColumnInfo(name = "reardriverwindow_value")
        var value: String? = null
    }

    @Generated("jsonschema2pojo")
    class RearPassWindowPos {
        @SerializedName("value")
        @Expose
        @ColumnInfo(name = "rearpasswindow_value")
        var value: String? = null
    }

    @SerializedName("driverWindowPosition")
    @Expose
    @Embedded
    var driverWindowPosition: DriverWindowPosition? = null

    @SerializedName("passWindowPosition")
    @Expose
    @Embedded
    var passWindowPosition: PassWindowPosition? = null

    @SerializedName("rearDriverWindowPos")
    @Expose
    @Embedded
    var rearDriverWindowPos: RearDriverWindowPos? = null

    @SerializedName("rearPassWindowPos2")
    @Expose
    @Embedded
    var rearPassWindowPos: RearPassWindowPos? = null
}

